config/clusters/2i2c.cluster.yaml

```diff
@@ -495,6 +495,48 @@ hubs:
                 - swalker
                 - shaolintl
               admin_users: *aup_users
+  - name: anu
+    domain: anu.pilot.2i2c.cloud
+    helm_chart: basehub
+    auth0:
+      connection: CILogon
+    config:
+      jupyterhub:
+        custom:
+          2i2c:
+            add_staff_user_ids_to_admin_users: true
+            add_staff_user_ids_of_type: "google"
+          homepage:
+            templateVars:
+              org:
+                name: The Australian National University
+                logo_url: https://upload.wikimedia.org/wikipedia/en/thumb/4/4c/Australian_National_University_%28emblem%29.svg/320px-Australian_National_University_%28emblem%29.svg.png
+                url: https://www.anu.edu.au/
+              designed_by:
+                name: 2i2c
+                url: https://2i2c.org
+              operated_by:
+                name: 2i2c
+                url: https://2i2c.org
+              funded_by:
+                name: ANU
+                url: https://www.anu.edu.au/
+        singleuser:
+          memory:
+            limit: 2G
+            guarantee: 2G
+          cpu:
+            guarantee: 1
+            limit: 2
+          image:
+            name: quay.io/mmcky/anu-emet2007
+            tag: b7937f446fe6
+        hub:
+          config:
+            Authenticator:
+              admin_users:
+                - matthew.mckay@anu.edu.au
+              username_pattern: '^(.+@2i2c\.org|.+@anu\.edu\.au|deployment-service-check)$'
```

Comment:
Do we need to set `allowed_users` here?

Reply:
No, we don't need the `allowed_users` list when `username_pattern` is used. Basically, `username_pattern` does the same thing like `allowed_users`, but using a regex, rather that a specific list that should be matched. Not sure what happens if you specify both.

Reply:
Ah great! Could you add an inline comment explaining that please?

Reply:
But since admins need to match the pattern to be allowed access, I imagine any user in the `allowed_users` list would have to match too. So, I believe `username_pattern` takes precedence over both.

Reply:
All I'm really after is clarity on when we do use `allowed_users` or not. Since we know it's a security risk for the majority of our hubs, but it can block access in some cases, e.g., GitHub Orgs/Teams Auth. So if we're _not_ explicitly listing `allowed_users`, I think it's a good idea to add a short comment explaining why so other engineers don't add it by mistake thinking their plugging a security leak :)

Reply:
:tada: excellent, I lacked comprehension about that as well and have become very cautious about anything with these settings, so this clarification in review + inline comment is very appreciated!

Reply:
Done! Thank you :D